osu.Game/Screens/OnlinePlay/Components/OnlinePlayBackgroundScreen.cs

```diff
@@ -16,65 +21,57 @@ namespace osu.Game.Screens.OnlinePlay.Components
     public abstract partial class OnlinePlayBackgroundScreen : BackgroundScreen
     {
         private CancellationTokenSource? cancellationSource;
-        private PlaylistItemBackground? background;
+        private Background? lastBackground;
+        private int? beatmapId;
 
         [BackgroundDependencyLoader]
         private void load()
         {
-            switchBackground(new PlaylistItemBackground(playlistItem));
+            loadNewBackground();
         }
 
-        private PlaylistItem? playlistItem;
-
         protected PlaylistItem? PlaylistItem
         {
-            get => playlistItem;
             set
             {
-                if (playlistItem == value)
+                if (beatmapId == value?.Beatmap.OnlineID)
                     return;
 
-                playlistItem = value;
+                beatmapId = value?.Beatmap.OnlineID;
 
-                if (LoadState > LoadState.Ready)
-                    updateBackground();
+                if (LoadState >= LoadState.Ready)
+                    loadNewBackground();
             }
         }
 
-        private void updateBackground()
+        private void loadNewBackground()
         {
-            Schedule(() =>
-            {
-                var beatmap = playlistItem?.Beatmap;
+            cancellationSource?.Cancel();
+            cancellationSource = new CancellationTokenSource();
 
-                string? lastCover = (background?.Beatmap?.BeatmapSet as IBeatmapSetOnlineInfo)?.Covers.Cover;
-                string? newCover = (beatmap?.BeatmapSet as IBeatmapSetOnlineInfo)?.Covers.Cover;
```

Comment:
This now dedupes on beatmap id (inside `PlaylistItem` setter) rather than cover resource. It does mean that the background will be reloaded if you select multiple beatmaps from the same set, but I don't think this is particularly important to consider - multi/playlists generally don't have multiple beatmaps from the same set in sequence.